tests/test_dms.py

```diff
@@ -98,6 +98,36 @@ def test_modify(self):
             [True, "dms.t2.medium", "mon:23:00-mon:23:59"],
         )
 
+    def test_replication_instance_kmskey(self):
+        session_factory = self.replay_flight_data("test_dms_kmskey")
+        p = self.load_policy(
+            {
+                "name": "dms-instance-kmskey",
+                "resource": 'dms-instance',
+                "filters": [
+                    {
+                        "type": "kms-key",
+                        "key": '"c7n:AliasMetadata".AliasName',
+                        "value": "alias/aws/dms",
+                        "op": "eq"
+                    }
+                ]
+            },
+            session_factory=session_factory,
+        )
+        resources = p.run()
+        client = session_factory().client("kms")
+        aliases = client.list_aliases()
```

Comment:
rather than listing all aliases just list the alias for the key arn on the resource ie. see KeyId parameter on this api call.